third_party/Makefile

```diff
@@ -0,0 +1,39 @@
+go-oniguruma:
+	DIR=$$(mktemp -d) && \
+	echo "$(abspath $@)" && \
+	cd $$DIR && \
+	git clone --branch v1.2.1 git@github.com:go-enry/go-oniguruma.git go-oniguruma && \
+	cd go-oniguruma && \
+	git apply $(abspath go-oniguruma.patch) && \
+	rm -rf "$(abspath $@)" && \
+	cp -a $$DIR"/go-oniguruma" "$(abspath $@)"
+
+oniguruma:
+	DIR=$$(mktemp -d) && \
+	echo "$(abspath $@)" && \
+	cd $$DIR && \
+	git clone --branch v6.9.10 git@github.com:kkos/oniguruma.git oniguruma && \
+	cd oniguruma && \
+	git apply $(abspath oniguruma.patch) && \
+	rm -rf "$(abspath $@)" && \
+	cp -a $$DIR"/oniguruma" "$(abspath $@)"
+
+.PHONY: update_go-oniguruma.patch
+update_go-oniguruma.patch:
+	cd go-oniguruma && \
+	git add . && \
+	git diff --staged > $(abspath go-oniguruma.patch)
```

Comment:
I don't think this calculates the diff that you want. This calculates the diff between the working copy's go-oniguruma and the checked-in go-oniguruma. But you want to calculate the diff between the working copy's go-oniguruma and the *upstream* go-oniguruma.

Reply:
Updated the command so now it compares directly with the remote tag which we use to clone :
```
	git fetch --tags origin && \
	git add . && \
	git diff --staged v1.2.1 > $(abspath go-oniguruma.patch)
```

Done.

Reply:
Update : As pointed out by @quentinmit, this command was relying on it still being a Git repo (having a `.git`). The assumption was that everything under `third_party` should not be a Git repo to avoid polluting the main repo. I fixed both things : 
- The copy command to `third_party` uses `$$DIR"/go-oniguruma"/*` to avoid copying any hidden file (e.g. `.git`).
- The `go-oniguruma.patch` rule now copies all files in `go-oniguruma/*` folder to a freshly cloned repo and executes "git diff" into `go-oniguruma.patch`.

https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/blob/3d39dcbed125ccd45522570fbc9cd146d9bf6f76/third_party/Makefile#L4-L53